internal/controllers/cryostat_controller_test.go

```diff
@@ -213,6 +211,33 @@ var _ = Describe("CryostatController", func() {
 					})
 				})
 			})
+		}
+		Context("successfully creates required resources", func() {
+			BeforeEach(func() {
+				t.objs = append(t.objs, t.NewCryostat())
+			})
+
+			expectSuccessful()
+		})
+		Context("with multiple namespaces", func() {
+			namespaces := []string{"test-one", "test-two"}
+			BeforeEach(func() {
+				for _, ns := range namespaces {
+					t.Namespace = ns
+					t.objs = append(t.objs, t.NewNamespace(), t.NewCryostat())
+				}
+			})
+
+			for _, ns := range namespaces {
+				ns := ns // capture value
```

Comment:
Not sure but I wonder what problem causes us to add this?

Reply:
This tripped me up too, but the problem is the `t.Namespace = ns` assignment, which occurs inside an anonymous function. This function is a callback that isn't executed synchronously. The value of the loop variable `ns` changes, and by the time the callback function is called, each invocation will use the latest value of `ns`. By creating a new variable inside the loop, this ensures that `ns` inside the function uses the intended value of `ns` at each iteration.

More info here: https://go.dev/doc/faq#closures_and_goroutines and here: https://onsi.github.io/ginkgo/#dynamically-generating-specs

Reply:
Wow that is something new today! Didnt even know such issue exists XD